python-checks/src/test/java/org/sonar/python/checks/FileHeaderCopyrightCheckTest.java

```diff
@@ -141,4 +141,20 @@ public void test_searchPattern_exception() {
     IllegalArgumentException e = Assert.assertThrows(IllegalArgumentException.class, () -> SubscriptionVisitor.analyze(check, context));
     Assertions.assertThat(e.getMessage()).isEqualTo("[FileHeaderCopyrightCheck] Unable to compile the regular expression: "+fileHeaderCopyrightCheck.headerFormat);
   }
+
+  @Test
+  public void shebangTest() {
+    var fileHeaderCopyrightCheck = new FileHeaderCopyrightCheck();
+    fileHeaderCopyrightCheck.headerFormat = "# Copyright FOO\n";
+    PythonCheckVerifier.verifyNoIssue("src/test/resources/checks/fileHeaderCopyright/shebangCopyright.py", fileHeaderCopyrightCheck);
+  }
+
+  @Test
+  public void shebangPatternTest() {
+    var fileHeaderCopyrightCheck = new FileHeaderCopyrightCheck();
+    fileHeaderCopyrightCheck.isRegularExpression = true;
+    fileHeaderCopyrightCheck.headerFormat = "# Copyright[ ]20[0-9]{2}";
+    PythonCheckVerifier.verifyNoIssue("src/test/resources/checks/fileHeaderCopyright/searchPatternShebangCopyright.py",
```

Comment:
I would also add a test where we use the shebang line, but the copyright header is noncompliant.